code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,674 @@
+				if (0 to 14)
+					icon_state = "nuclearbomb3"
+					quiet  = !quiet
+					if(!quiet)
+						return
+					else
+						playsound(loc, 'sound/machines/engine_alert2.ogg', 100, 0)
+
+/obj/machinery/construction_nuke/interact(mob/user)
+	user.set_machine(src)
+	var/list/dat = list()
+	dat +="<div class='statusDisplay'>"
+	dat += "Timer: [get_time_left()] seconds<br>"
+	dat += "</div>"
+	dat += "<b><u>Detonation Payload</u>: <A href='?src=\ref[src];action=payload'>[payload]</A></b><br><br>"
+	dat += "<A href='?src=\ref[src];action=set'>Set Timer</A><br>"
+	dat += "<A href='?src=\ref[src];action=anchor'>[anchored ? "Anchored" : "Not Anchored"]</A><br>"
+	dat += "<A href='?src=\ref[src];action=safety'>[safety ? "Safety On" : "Safety Off"]</A><br><br>"
+	dat += "<b><A href='?src=\ref[src];action=activate'>[bomb_set ? "DEACTIVATE" : "ACTIVATE"]</A><b><br>"
+	var/datum/browser/popup = new(user, "vending", "Construction Nuke", 300, 275)
+	popup.set_content(dat.Join())
+	popup.open()
+
+
+/obj/machinery/construction_nuke/Topic(href, href_list)
+	if(..())
+		return
+	switch(href_list["action"])
+		if ("payload")
+			set_payload()
+			updateUsrDialog()
+		if ("set")
+			set_timer()
+			updateUsrDialog()
+		if ("anchor")
+			set_anchor()
+			updateUsrDialog()
+		if ("safety")
+			set_safety()
+			updateUsrDialog()
+		if ("activate")
+			set_active()
+			updateUsrDialog()
+
+/obj/machinery/construction_nuke/proc/set_payload()
+	playsound(loc, 'sound/machines/terminal_prompt.ogg', 75, 1)
+	if(timing || bomb_set)
+		to_chat(usr, "<span class='danger'>Error: Payload cannot be altered while the device is armed.</span>")
+		playsound(loc, 'sound/machines/defib_failed.ogg', 75, 1)
+		return
+	payload = input(usr, "Choose your Payload", "Payload:") as null|anything in possible_payloads
+	if (!src || QDELETED(src))
+		return
+	playsound(loc, 'sound/machines/terminal_prompt_confirm.ogg', 75, 1)
+	switch(payload)
+		if("plasteel")
+			payload_wall = /turf/closed/wall/r_wall
+			payload_floor = /turf/open/floor/engine
+		if("wood")
+			payload_wall = /turf/closed/wall/mineral/wood
+			payload_floor = /turf/open/floor/wood
+		if("sand")
+			payload_wall = /turf/closed/wall/mineral/sandstone
+			payload_floor = /turf/open/floor/plating/beach/sand
+		if("ice")
+			payload_wall = /turf/closed/wall/ice
+			payload_floor = /turf/open/floor/plating/ice
+		if("mining")
+			payload_wall = /turf/closed/wall/shuttle/survival/pod
+			payload_floor = /turf/open/floor/plating/asteroid/basalt/lava
+		if("desolation")
+			payload_wall = /turf/closed/wall/rust
+			payload_floor = /turf/open/floor/fakespace
+		if("bluespace")
+			payload_wall = /turf/closed/wall/mineral/titanium
+			payload_floor = /turf/open/floor/bluespace
+		if("safety")
+			payload_wall = /turf/closed/wall/r_wall
+			payload_floor = /turf/open/floor/noslip
+		if("bananium")
+			payload_wall = /turf/closed/wall/mineral/clown
+			payload_floor = /turf/open/floor/mineral/bananium
+		else
```

Comment:
>Clown

No it doesnt. I actually fixed this in my original PR (which required me to change the clown shuttle) but atomization killed any chance of me fixing it again.